lib/bundle/brew_installer.rb

```diff
@@ -14,21 +14,35 @@ def initialize(name, options = {})
       @full_name = name
       @name = name.split("/").last
       @args = options.fetch(:args, []).map { |arg| "--#{arg}" }
+      @unlink = options.fetch(:unlink, false)
+      @stop_service = options.fetch(:stop_service, false)
       @restart_service = options.fetch(:restart_service, false)
     end
 
     def run
-      install_or_upgrade and restart_service!
+      install_change_state! && service_change_state!
     end
 
-    def install_or_upgrade
+    def unlink?
+      @unlink
+    end
+
+    def install_change_state!
       if installed?
-        upgrade!
+        if unlink?
+          unlink!
+        else
+          upgrade!
+        end
       else
         install!
       end
     end
```

Comment:
So for `brew "foo", unlink: true`, we will have:
* Unlink it if it's installed.
* Install (and link it) if it's not installed.

This is very counter-intuitive. From what I'm understanding, what you really need is an unlink-if-conflict feature.